File: 0179-largest-number/0179-largest-number.cpp
class Solution {
public:
    static bool cmp(string& a, string& b) {
        return a+b > b+a;
    }
    string largestNumber(vector<int>& nums) {
        vector<string> numstring;
        for (auto n : nums) {
            numstring.push_back(to_string(n));
        }

        sort(numstring.begin(), numstring.end(), cmp);
        string answer;

        for (auto s : numstring) {
            answer += s;
        }
        if (answer[0] == '0') return "0";
        return answer;
    }
};

File: 0179-largest-number/0179-largest-number.py
class Solution:
    def largestNumber(self, nums: List[int]) -> str:
        nums = [str(n) for n in nums]
        nums.sort(key = lambda x:x*10, reverse=True)
        largest = ""
        for n in nums:
            largest += n
        if largest[0] == "0":
            return "0"
        return largest